.github/workflows/mypy.yaml

```diff
@@ -0,0 +1,32 @@
+name: mypy
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - synchronize
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  ci:
+    runs-on: ubuntu-20.04
+    steps:
+      # Checkout the code
+      - uses: actions/checkout@v2
+      - uses: jpetrucciani/mypy-check@master
+        with:
+          requirements: 1.6.0
+          python_version: 3.11
```

Comment:
it would be great to test against Python `3.8`, which does not have the new style annotations to ensure backwards compatibility with previous Python versions. WDYT?

Reply:
Yeah makes sense. I'll move this back to 3.8